### PR TITLE
TPv2 Update Angular and make compilation stricter

### PR DESCRIPTION
**Pull request by ocket8888**
This PR updates Angular to version 13 (13.0.3 at the time of this writing) for TPv2, and adds some options to get type-checking in templates, as well as adding some more strict type checking generically. Mostly changes that needed to made to make it compile came from the strict template typing. And the fact that it was using a "user" as the same thing as a "current user", which unfortunately cannot be safely done (see #6299).

This also fixes some broken tests from the module restructure done in #6263, and squashes an error about Chart JS being bulky and causing optimization problems. That's all true, but we know about it, so there's no reason to see a warning on every compilation.

Also, the new version of Angular allows the use of ES2020 in source, and because IE is dead and Edge is now Chromium-based, our output has moved up to ES2017, which is nice.

<hr/>

## Which Traffic Control components are affected by this PR?
- Traffic Portal (experimental v2)

## What is the best way to verify this PR?
Make sure the unit and end-to-end tests pass. Also probably play around with the UI a bit, because our end-to-end tests only cover the login page.

## If this is a bugfix, which Traffic Control versions contained the bug?
- master, but who cares, it's experimental

## PR submission checklist
- [x] This PR has tests
- [ ] This PR has documentation
- [ ] This PR has a CHANGELOG.md entry
- [x] This PR **DOES NOT FIX A SERIOUS SECURITY VULNERABILITY**

### CODE REVIEW

**Comment by ocket8888** [comment]
I know this PR looks big, but 49.k lines of changes are coming from just package-lock.json.

File: experimental/traffic-portal/package.json [hunk 1]
```diff
@@ -22,60 +22,59 @@
...
+    "@angular-eslint/template-parser": "13.0.1",
+    "@angular/cli": "~13.0.3",
+    "@angular/compiler-cli": "~13.0.2",
     "@types/argparse": "^2.0.2",
-    "@types/chart.js": "^2.9.32",
+    "@types/chart.js": "^2.9.34",
     "@types/express": "^4.17.0",
     "@types/jasmine": "~3.6.0",
     "@types/jasminewd2": "~2.0.3",
     "@types/node": "^12.11.1",
```
Comment (shamrickus) [inline experimental/traffic-portal/package.json:69]: Should be upgrade to 14?

Comment (ocket8888) [reply experimental/traffic-portal/package.json:69]: yeah, it seems like it should.

**Review by shamrickus — APPROVED** [review]
LGTM

**Pull request merged**